### ISSUE TITLE
Test the NEW TOKEN frame

### ISSUE THREAD

**Issue by huitema** [private-octopus/picoquic#363]
The current code does not use the NEW TOKEN frame. If address verification is requested, all connections attempts result in a retry. The suggestion is to:

1) Define a new token structure in which the connection ID can be missing;
2) Accept connections carrying that new token, trigger retry for the other connections;
3) Keep generating a connection-ID carrying token when sending a retry;
4) Send a NEW TOKEN with null length CID after each successful connection;
5) Manage a store of new token on the client.

This will allow dynamic usage of the "retry" process as a reaction to overload, without too much penalty for the "good clients".

**Comment by huitema** [comment]
Need to postpone until draft 17. There may well be a need to ensure that new token -- and tokens in general -- are used only once. One plausible outcome is to embed a server CID inside the new token, but this will require some special code in the parsing. In any case, one needs to insert a time to live and a peer address.

**Comment by huitema** [comment]
Done in PT #503
